lexers/o/org.go

```diff
@@ -0,0 +1,80 @@
+package o
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Markdown lexer.
+var Org = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "Org Mode",
+		Aliases:   []string{"org", "orgmode"},
+		Filenames: []string{"*.org"},
+		MimeTypes: []string{"text/org"}, // https://lists.gnu.org/r/emacs-orgmode/2017-09/msg00087.html
+	},
+	Rules{
+		"root": {
+			{`^#\s.*$`, Comment, nil},
+			// Headings
+			{`^(\*+)( COMMENT)( .*)$`, ByGroups(GenericHeading, NameBuiltin, Text), nil},
+			{`^(\*\*+)( COMMENT)( .*)$`, ByGroups(GenericSubheading, NameBuiltin, Text), nil},
+			{`^(\*+)( DONE)( .*)$`, ByGroups(GenericHeading, NameBuiltin, Text), nil},
+			{`^(\*\*+)( DONE)( .*)$`, ByGroups(GenericSubheading, NameBuiltin, Text), nil},
+			{`^(\*+)( TODO)( .*)$`, ByGroups(GenericHeading, Error, Text), nil},
+			{`^(\*\*+)( TODO)( .*)$`, ByGroups(GenericSubheading, Error, Text), nil},
+			{`^(\*)( .+?)( :[a-zA-Z0-9_@:]+:)()$`, ByGroups(GenericHeading, GenericStrong, GenericEmph, Text), nil}, // Level 1 heading with tags
+			{`^(\*)( .+)$`, ByGroups(GenericHeading, GenericStrong), nil}, // // Level 1 heading with NO tags
+			{`^(\*\*+)( .+?)( :[a-zA-Z0-9_@:]+:)()$`, ByGroups(GenericSubheading, Text, GenericEmph, Text), nil}, // Level 2+ heading with tags
+			{`^(\*\*+)( .+)$`, ByGroups(GenericSubheading, Text), nil}, // Level 2+ heading with NO tags
+			// Checkbox lists
+			{`^(\s*)([+-] )(\[[ X]\])( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			{`^( +)(\* )(\[[ X]\])( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			// Definition lists
+			{`^(\s*)([+-] )([^ \n]+ ::)( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			{`^( +)(\* )([^ \n]+ ::)( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			// Unordered lists
+			{`^(\s*)([+-] )(.+)$`, ByGroups(Text, Keyword, UsingSelf("inline")), nil},
+			{`^( +)(\* )(.+)$`, ByGroups(Text, Keyword, UsingSelf("inline")), nil},
+			// Ordered lists
+			{`^(\s*)([0-9]+[.)])( \[@[0-9]+\])( .+)$`, ByGroups(Text, Keyword, GenericEmph, UsingSelf("inline")), nil},
+			{`^(\s*)([0-9]+[.)])( .+)$`, ByGroups(Text, Keyword, UsingSelf("inline")), nil},
+			// Blocks
+			{`^(\s*#\+begin_src)( [^ \n]+)(.*\n)([\w\W]*?)(^\s*#\+end_src$)`,
+				UsingByGroup(
+					internal.Get,
+					2, 4,
+					Comment, Comment, Comment, Text, Comment,
+				),
+				nil,
+			},
+			{`^(\s*#\+begin_\w+\n)([\w\W]*?)(^\s*#\+end_\w+$)`, ByGroups(Comment, Text, Comment), nil},
+			// Keywords
+			// Fri Jul 27 00:36:20 EDT 2018 - kmodi
+			// Unable to get the below to work.
+			// {`^(#\+options)(:)(((\s)([^:\n]+?)(:[^:\s\n]+))+)()$`, ByGroups(GenericStrong, Text, None, None, Text, GenericEmph, Text, Text), nil}, // Org keyword #+options
```

Comment:
I would like to parse a line like:

```
#+options: foo1:val1 foo2:val2 ..
```

Here we don't know how many fooN:valN pairs we will have.. So I tried to match that with `(((\s)([^:\n]+?)(:[^:\s\n]+))+)`.. trying to mean that we have at least one of those fooN:valN pairs. But I couldn't figure out how to set a different category for each for those subgroups.

Reply:
You'll need to introduce a new lexer state and use `Push("options")`. You can read the [Pygments docs](http://pygments.org/docs/lexerdevelopment/) for more information; Chroma is identical in this regard.

Reply:
I might have to postpone this feature for a future PR until I study up on those docs and `Push` examples from other lexers in chroma. Thanks.